webapp/src/utils/fileUtils.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Triggers the download of a file with the given data and name.
+ *
+ * @param fileData - The data of the file to be downloaded.
+ * @param fileName - The name of the file to be downloaded.
+ */
+export function downloadFile(fileData: BlobPart, fileName: string) {
+  const link = document.createElement("a");
+  link.href = URL.createObjectURL(new Blob([fileData]));
```

Comment:
you can add the mimeType `{ type: "application/vnd. ms-excel }` (excel files only may be adjusted if the processDownload is intended for a global usage)